packages/web/docs/src/pages/docs/gateway/other-features/plugins.mdx

```diff
@@ -0,0 +1,628 @@
+---
+description: Add feature or third party integrations with Hive Gateway plugin system.
+---
+
+import { Callout } from '@theguild/components'
+
+# Plugins
+
+Hive Gateway is built with an highly modular architecture. The core is as small as possible, and
+most of the feature are built around a plugin system.
+
+A plugin can hook into each important steps of the lifecycle of the gateway, including the graphql
+request execution.
+
+The gateway is built using Graphql Yoga (a GraphQL HTTP server) and Envelop (a GraphQL execution
+orchestrator), which are also highly modular technologies. Hive Gateway is extending Yoga and
+Envelop plugin system by providing gateway specific hooks.
+
+This means that if a plugin exists in multiple variants, you should follow this priority order:
```

Comment:
It's because if they want to use a plugin, it can exists in multiple variants. The Prometheus plugin is a good example of this. You have to choose the correct variant to have access to the best of it.

In the other hand, we have plugins that exists only in the envelop variant, like Sentry. So we have to some way say that you can use Envelop or Yoga plugins too.

Reply:
Maybe this is also a technical details that we don't need them to be aware of?
why not create a gateway variant for any plugin users might need?

Reply:
I do agree. IMO, we should just have one package for a plugin. Then, it should be up to the plugin to handle the fact that some hooks are optionally called, depending on where they are used.

Yet, it's not the case today, so what do we do ? Should we document this point until we refactor plugins ?